From 6f9ad2cd03091ad4fed7528004bb3168e145ce86 Mon Sep 17 00:00:00 2001
From: Mathias Kresin <dev@kresin.me>
Date: Fri, 7 Apr 2017 23:52:27 +0200
Subject: [PATCH 67/69] ramips: prepare ZBT-WG3526 for hardware variants

The ZBT-WG3526 is available with 16 or 32 MByte of flash. Rename the
current supported 16MByte version to indicate which flash size variant
is supported.

Signed-off-by: Mathias Kresin <dev@kresin.me>

Conflicts (Resolved):
	target/linux/ramips/image/mt7621.mk

Signed-off-by: Vincent Wiemann <me@bibbl.com>

further adjustments for backport to lede-17.01 branch

Signed-off-by: Andreas Ziegler <github@andreas-ziegler.de>
---
 target/linux/ramips/base-files/lib/ramips.sh                   | 4 ++--
 target/linux/ramips/dts/{ZBT-WG3526.dts => ZBT-WG3526-16M.dts} | 2 +-
 target/linux/ramips/image/mt7621.mk                            | 9 +++++----
 3 files changed, 8 insertions(+), 7 deletions(-)
 rename target/linux/ramips/dts/{ZBT-WG3526.dts => ZBT-WG3526-16M.dts} (83%)

diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index b1091f66b2..8e27a79fc1 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -637,8 +637,8 @@ ramips_board_detect() {
 	*"ZBT-WG2626")
 		name="zbt-wg2626"
 		;;
-	*"ZBT-WG3526")
-		name="zbt-wg3526"
+	*"ZBT-WG3526 (16M)")
+		name="zbt-wg3526-16M"
 		;;
 	*"ZBT-WR8305RT")
 		name="zbt-wr8305rt"
diff --git a/target/linux/ramips/dts/ZBT-WG3526.dts b/target/linux/ramips/dts/ZBT-WG3526-16M.dts
similarity index 83%
rename from target/linux/ramips/dts/ZBT-WG3526.dts
rename to target/linux/ramips/dts/ZBT-WG3526-16M.dts
index c361bdd827..fb64450207 100644
--- a/target/linux/ramips/dts/ZBT-WG3526.dts
+++ b/target/linux/ramips/dts/ZBT-WG3526-16M.dts
@@ -3,7 +3,7 @@
 #include "ZBT-WG3526.dtsi"
 
 / {
-	model = "ZBT-WG3526";
+	model = "ZBT-WG3526 (16M)";
 };
 
 &firmware {
diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 8218bea844..cabc86b28e 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -226,13 +226,14 @@ define Device/zbt-wg2626
 endef
 TARGET_DEVICES += zbt-wg2626
 
-define Device/zbt-wg3526
-  DTS := ZBT-WG3526
+define Device/zbt-wg3526-16M
+  DTS := ZBT-WG3526-16M
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
-  DEVICE_TITLE := ZBT WG3526
+  SUPPORTED_DEVICES += zbt-wg3526
+  DEVICE_TITLE := ZBT WG3526 (16MB flash)
   DEVICE_PACKAGES := kmod-usb3 kmod-usb-ledtrig-usbport kmod-ata-core kmod-ata-ahci
 endef
-TARGET_DEVICES += zbt-wg3526
+TARGET_DEVICES += zbt-wg3526-16M
 
 # FIXME: is this still needed?
 define Image/Prepare
-- 
2.16.2

